Biasing/include/Biasing/Utility/StepPrinter.h

```diff
@@ -21,27 +21,36 @@ class StepPrinter : public simcore::UserAction {
    * @param[in] parameters the parameters used to configure this
    *      UserAction.
    */
-  StepPrinter(const std::string& name, framework::config::Parameters& parameters);
+  StepPrinter(const std::string& name,
+              framework::config::Parameters& parameters);
 
   /// Destructor
-  ~StepPrinter();
+  virtual ~StepPrinter() = default;
 
   /**
    * Stepping action called when a step is taken during tracking of
    * a particle.
    *
    * @param[in] step Geant4 step
    */
-  void stepping(const G4Step* step) final override;
+  void stepping(const G4Step* step) override;
 
   /// Retrieve the type of actions this class defines
-  std::vector<simcore::TYPE> getTypes() final override {
+  std::vector<simcore::TYPE> getTypes() override {
     return {simcore::TYPE::STEPPING};
   }
 
  private:
   /// The track ID to filter on
   int trackID_{-9999};
+  std::string processName_{"UNDEFINED"};
+  int depth_{0};
+  std::unordered_map<int,int> trackParents_{};
+
+
+  /// Check if the given track is a descendent of the track we are interested in
+  /// up to a certain depth
+  bool isDescendent(const G4Track* track) const;
```

Comment:
This functionality is already present (although inaccessible) within the `TrackMap`.

https://github.com/LDMX-Software/SimCore/blob/191f2eccfefb20f293f9c2e8ab5e89ec3da1d9aa/include/SimCore/TrackMap.h#L128-L129

Perhaps we add a function to `TrackMap` so that we don't have two copies of the ancestry tree?
```c++
bool TrackMap::isDescendent(int potential_ancestor_id, int track_id, int max_depth);
```

`TrackMap::findIncident` does something similar although instead of depth, it looks for first ancestor that was born outside of the calorimeter region. One can retrieve the `TrackMap` via
```cpp
g4user::TrackingAction::get()->getTrackMap()
```
We could add this retreival as a protected member of the `UserAction` class like [a few other helpers](https://github.com/LDMX-Software/SimCore/blob/191f2eccfefb20f293f9c2e8ab5e89ec3da1d9aa/include/SimCore/UserAction.h#L158-L177).

**Moving the implementation into `TrackMap` is not necessary**, but I am slightly concerned that this copying of the ancestry tree could leak into code that is more performance critical.